### PR TITLE
UI: Provide Data step

### PR DESCRIPTION
**Pull request by kpyszkowski**
Ref: #2 
Depends on: #12 

This pull request introduces redesigned Provide Data minting flow step

localhost_3000_tBTC_mint (1)

### CODE REVIEW

File: src/theme/Checkbox.ts [hunk 1]
```diff
@@ -3,21 +3,36 @@ import { PartsStyleFunction } from "@chakra-ui/theme-tools"
...
+      borderRadius: "md",
       _checked: {
-        borderColor: "brand.500",
+        borderColor: "hsl(182, 100%, 70%)",
         _active: {
-          borderColor: "brand.500",
+          borderColor: "hsl(182, 100%, 70%)",
         },
-        backgroundColor: "brand.500",
+        bg: "hsl(182, 100%, 70%)",
```
Comment (michalsmiarowski) [inline src/theme/Checkbox.ts:22]: Isn't it the same color as `#66F9FF`, which is `branch.100` in our color theme?

If yes, then let's change all the occurences of `hsl(182, 100%, 70%)` in our app to `brand.100`.

File: src/theme/Checkbox.ts [hunk 2]
```diff
@@ -3,21 +3,36 @@ import { PartsStyleFunction } from "@chakra-ui/theme-tools"
...
       },
+      _focus: {
+        boxShadow: "0 0 0 3px hsla(182, 100%, 70%, 20%)",
+      },
     },
     icon: {
-      color: "white",
+      w: 3,
+      h: 3,
+      color: "#121212",
```
Comment (michalsmiarowski) [inline src/theme/Checkbox.ts:35]: Btw, is there any specific reason you sometimes use `hsl` and sometimes `#<color_name>`? I think it would be better to stick to one syntax.

**Review by michalsmiarowski — CHANGES_REQUESTED** [review]
Overall looking good 🔥 Left some comments too look at before the merge

File: src/components/Forms/FormikInput.tsx [hunk 3]
```diff
@@ -4,9 +4,9 @@ import {
   FormControlProps,
   FormLabel,
   Input,
-  Stack,
   useColorModeValue,
```
Comment (michalsmiarowski) [inline src/components/Forms/FormikInput.tsx:7]: Also, there are lot of unused imports and variables across the app after your changes. Do you mind clearing them out and see if there are any unused imports/variables in each changed file?

**Review by michalsmiarowski — CHANGES_REQUESTED** [review]

File: src/theme/Checkbox.ts (hunk 2 above)
Comment (kpyszkowski) [reply src/theme/Checkbox.ts:35]: Definitely. Sadly we don't have the color palette (yet? 🤔) so I wasn't really strict with the color values.

There are a lot of shades of gray in designs, it's difficult to keep up the pace without the color palette XD
Personally I'd like to stick to HSL(A) color representation model. In my opinion it's much more human readable than hexadecimal representation.

If you agree just please let me know if you spot any occurance of hexadecimal color definition while reviewing my code so I can transform it into HSL.

File: src/theme/Checkbox.ts (hunk 1 above)
Comment (kpyszkowski) [reply src/theme/Checkbox.ts]: Ref commit: ad674a24

File: src/pages/tBTC/Bridge/components/BridgeProcessCardTitle.tsx [hunk 4]
```diff
@@ -1,50 +1,90 @@
...
-      onPreviousStepClick: (previosuStep: MintingStep) => void
-    }
-  | {
-      previousStep?: never
-      onPreviousStepClick?: never
-    }
-
-type Props = CommonProps & ConditionalProps
+const TOTAL_STEPS = Object.keys(MintingStep).length - 1
+// -1 because we don't count the success step
```
[the remark this answers is not in the document]
Comment (kpyszkowski) [reply src/pages/tBTC/Bridge/components/BridgeProcessCardTitle.tsx]: Ref commit: 2a8cf54b

File: src/pages/tBTC/Bridge/Minting/ProvideData.tsx [hunk 5]
```diff
@@ -179,17 +192,33 @@ export const ProvideDataComponent: FC<{
...
-        loadingText={"Generating deposit address..."}
+        loadingText={"Generating Deposit Address..."}
         type="submit"
         form="tbtc-minting-data-form"
         isFullWidth
         data-ph-capture-attribute-button-name={`Generate Deposit Address (Deposit flow)`}
+        iconSpacing={2.5}
+        rightIcon={<Icon as={ChevronRightIcon} w={5} h={5} />}
       >
         Generate Deposit Address
```
[the remark this answers is not in the document]
Comment (kpyszkowski) [reply src/pages/tBTC/Bridge/Minting/ProvideData.tsx:226]: Ref commit: c21d803b

File: src/pages/tBTC/Bridge/Minting/ProvideData.tsx [hunk 6]
```diff
@@ -179,17 +192,33 @@ export const ProvideDataComponent: FC<{
...
+          (Recommended)
+        </Text>
+        <TooltipIcon
+          color="#888"
+          label="The file is extremely important in case you need to make a fast recovery.
+This file is important in the rare case of fund recovery.
+Keep it until you have received your tBTC token. One deposit, one receipt.
+This file contains a BTC recovery address, a wallet public key, a refund public key and a refund lock time of your deposit."
+          ml={3}
+        />
```
[the remark this answers is not in the document]
Comment (kpyszkowski) [reply src/pages/tBTC/Bridge/Minting/ProvideData.tsx]: In fact breaking strings that way is incorrect. The only "breakable" strings in JS are template strings but it results with line breaks in output strings. 
In plain JS/TS such regular string breaks will result as an error. In JSX it's allowed (but not recommended) since it's only a syntactic sugar and it's transpiled to JS though so all the line breaks will be removed by the transpiler. 

The only reasonable way to fit such long strings in 80 characters print width would be to break it into separate strings and concatenate it with addition operator but [it's out of Prettiers scope](https://prettier.io/docs/en/rationale#what-prettier-is-_not_-concerned-about) and also considered as incorrect.

To sum up - such _lenghty_ string values should be treated as an exception in terms of 80 characters print width - just like Prettier does. I applied your request but it needs to be brainstormed - in my opinion (and many others) it's considered incorrect and I had to do it manually.

Ref commit: 93ac29c1

File: src/pages/tBTC/Bridge/Minting/ProvideData.tsx [hunk 7]
```diff
@@ -148,29 +154,36 @@ export const ProvideDataComponent: FC<{
...
-      <BridgeProcessCardSubTitle
-        stepText="Step 1"
-        subTitle="Generate a Deposit Address"
+    <VStack mx={{ base: 0, lg: 10 }} align="stretch" spacing={8}>
+      <BridgeProcessCardTitle
+        number={1}
+        title="Deposit Address"
+        description="Based on these two addresses, the system will generate for you a unique BTC deposit address. There is no minting limit."
+        afterDescription={<Badge variant="subtle">Action OFF-Chain</Badge>}
       />
```
[the remark this answers is not in the document]
Comment (kpyszkowski) [reply src/pages/tBTC/Bridge/Minting/ProvideData.tsx]: Ref commit: 93ac29c1
Explained above

File: src/pages/tBTC/Bridge/Minting/MintingFlowRouter.tsx [hunk 8]
```diff
@@ -26,6 +26,10 @@ const MintingFlowRouterBase = () => {
     useIsTbtcSdkInitializing()
 
   const onPreviousStepClick = (previousStep?: MintingStep) => {
+    if (!previousStep) {
+      const previousStepIndex = mintingSteps.indexOf(mintingStep) - 1
+      previousStep = mintingSteps[previousStepIndex] as MintingStep
+    }
```
[the remark this answers is not in the document]
Comment (kpyszkowski) [reply src/pages/tBTC/Bridge/Minting/MintingFlowRouter.tsx:36]: Ref commit: b0a74d79

File: src/pages/tBTC/Bridge/Minting/MintingFlowRouter.tsx [hunk 9]
```diff
@@ -9,13 +9,13 @@ import { MakeDeposit } from "./MakeDeposit"
...
 import { useModal } from "../../../../hooks/useModal"
 import { ModalType } from "../../../../enums"
-import { BridgeContractLink } from "../../../../components/tBTC"
-import { BridgeProcessCardTitle } from "../components/BridgeProcessCardTitle"
 import { useRemoveDepositData } from "../../../../hooks/tbtc/useRemoveDepositData"
 import { useAppDispatch } from "../../../../hooks/store"
 import { tbtcSlice } from "../../../../store/tbtc"
 import { useIsTbtcSdkInitializing } from "../../../../contexts/ThresholdContext"
 
+const mintingSteps = Object.values(MintingStep)
```
[the remark this answers is not in the document]
Comment (kpyszkowski) [reply src/pages/tBTC/Bridge/Minting/MintingFlowRouter.tsx]: The major drawback of this solution is you have two places to maintain when doing some changes - when you add new step you have to add another element to this array separately. Under the hood enums are transpiled to key-value objects. 
I can't see any single reason to reorder elements alphabetically, especially if steps in application meaningfully appears in specific order so I don't get this argument 😄 
Nevertheless I added your proposal - if we already have such variable it should be utilised. I didn't see it before.

Ref commit: e3246ec0

File: src/components/TooltipIcon.tsx [hunk 10]
```diff
@@ -9,39 +8,33 @@ import {
...
-        borderRadius="0.25rem"
-        boxShadow={boxShadow}
+        color="black"
+        rounded="xl"
+        w="full"
+        maxW="352px"
       >
-        <PopoverArrow bg={bgColor} />
-        <BodySm as={PopoverBody} color={textColor}>
+        <PopoverArrow bg="#66f9ff" />
```
[the remark this answers is not in the document]
Comment (kpyszkowski) [reply src/components/TooltipIcon.tsx]: Ref commit: 259f9ae9

File: src/components/TooltipIcon.tsx [hunk 11]
```diff
@@ -9,39 +8,33 @@ import {
...
-            {...iconProps}
-          />
-        </span>
+        <Flex display="inline-flex" as="span">
+          <Icon as={InformationIcon} w={4} h={4} {...iconProps} />
+        </Flex>
       </PopoverTrigger>
       <PopoverContent
-        bg={bgColor}
+        bg="#66f9ff"
```
[the remark this answers is not in the document]
Comment (kpyszkowski) [reply src/components/TooltipIcon.tsx]: Ref commit: 259f9ae9

File: src/components/Forms/FormikInput.tsx (hunk 3 above)
Comment (kpyszkowski) [reply src/components/Forms/FormikInput.tsx]: Ref commit: 168a4cc

**Review by kpyszkowski — COMMENTED** [review]
Addressed comments

File: src/pages/tBTC/Bridge/components/BridgeProcessCardTitle.tsx [hunk 12]
```diff
@@ -1,50 +1,81 @@
...
-
-type Props = CommonProps & ConditionalProps
+const TOTAL_STEPS = mintingSteps.length - 1
+// -1 because we don't count the success step
 
-export const BridgeProcessCardTitle: FC<Props> = ({
-  previousStep,
+export const BridgeProcessCardTitle: FC<
+  {
+    number: number
```
Comment (michalsmiarowski) [inline src/pages/tBTC/Bridge/components/BridgeProcessCardTitle.tsx:20]: I feel like I've added this comment before but somehow I can't find it 🤔 Maybe it was deleted due to refresh or something, so I'll write it again:

Can we change the name of `number` property? I feel like it's too generic and doesn't say anything about what that property is doing and what this number is for. What about `stepNumer` or just `step`, since it is displayed next to `TOTAL_STEPS` number?

File: src/theme/Checkbox.ts (hunk 2 above)
Comment (michalsmiarowski) [reply src/theme/Checkbox.ts]: I don't mind. We can use `hsl` although I think that we would have to convert in hex color provided by designers to a HSL, which is an additional step to do 😅 Unless we can convert it straight in Figma?

File: src/pages/tBTC/Bridge/components/BridgeProcessCardTitle.tsx (hunk 12 above)
Comment (kpyszkowski) [reply src/pages/tBTC/Bridge/components/BridgeProcessCardTitle.tsx:20]: Ref commit: 6e2f421a

**Review by kpyszkowski — COMMENTED** [review]
Addressed review comments

File: src/pages/tBTC/Bridge/Minting/ProvideData.tsx
Comment (michalsmiarowski) [inline src/pages/tBTC/Bridge/Minting/ProvideData.tsx:1]: Non-blocking:

We should adjust the background image the way it is done in designs. Right now, in some places, the background is too light resulting with some issues to read some texts.

Design:
image

Implementation:
image

Note how hard it is to read the text below Generate Deposit Address button

This should be addresssed in separate PR (unless it's an easy fix)

File: src/pages/tBTC/Bridge/Minting/ProvideData.tsx [hunk 13]
```diff
@@ -148,29 +155,39 @@ export const ProvideDataComponent: FC<{
...
   return (
-    <Box mx={{ base: 0, lg: 10 }}>
-      <BridgeProcessCardTitle onPreviousStepClick={onPreviousStepClick} />
-      <BridgeProcessCardSubTitle
-        stepText="Step 1"
-        subTitle="Generate a Deposit Address"
+    <VStack mx={{ base: 0, lg: 10 }} align="stretch" spacing={8}>
+      <BridgeProcessCardTitle
+        step={1}
+        title="Deposit Address"
```
Comment (michalsmiarowski) [inline src/pages/tBTC/Bridge/Minting/ProvideData.tsx:184]:
```suggestion
        title="Provide a Deposit Address"
```

File: src/pages/tBTC/Bridge/Minting/ProvideData.tsx [hunk 14]
```diff
@@ -179,17 +196,44 @@ export const ProvideDataComponent: FC<{
...
+        <TooltipIcon
+          color="#888"
+          label="
+            The file is extremely important in case you need to make a fast 
+            recovery. This file is important in the rare case of fund recovery.
+            Keep it until you have received your tBTC token. One deposit, one 
+            receipt. This file contains a BTC recovery address, a wallet public 
+            key, a refund public key and a refund lock time of your deposit.
+          "
+          ml={3}
```
Comment (michalsmiarowski) [inline src/pages/tBTC/Bridge/Minting/ProvideData.tsx:213]: Let's remove the `ml` property from `TooltipIcon` and add `mr` to recommended `Text` above. The reason for that is the `PopoverContent` is slighty off from the icon right now:

image

File: src/pages/tBTC/Bridge/Unmint.tsx [hunk 15]
```diff
@@ -73,11 +72,7 @@ const UnmintFormPage: PageComponent = ({}) => {
 
   return (
     <>
-      <BridgeProcessCardTitle bridgeProcess="unmint" />
-      <BridgeProcessCardSubTitle
-        stepText="Step 1"
-        subTitle="Unmint your tBTC tokens"
-      />
+      <BridgeProcessCardTitle step={1} title="Unmint your tBTC tokens" />
```
Comment (michalsmiarowski) [inline src/pages/tBTC/Bridge/Unmint.tsx:75]: It just popped out to me now, that `BridgeProcessCardTitle` is not reusable enough to use it in `Unmint` or `ResumeDeposit`. As of now it looks like we don't have the designs for those, but we will probably implement `Unmint` (not sure about `ResumeDeposit`). For `Unmint` though, the TOTAL_STEPS count will not be the same as in minting flow, so I think that we should either (from worst to best, pick one, I recommend picking 3):

1. Rename `BridgeProcessCardTitle` to `MintingProcessCardTitle`, but I fell like we will want to use some of it's code to Unmint and ResumeDeposit in the future. 
2. Make `steps` an optional property and allow to add `totalSteps` optionally too. We will display the `{step}/{TOTAL_STEPS}` thing only when `steps` AND `totalSteps` are provided
3. (this one would be the best IMO) Forget about `steps` / `totalSteps` whatsoever and create a new property called `badgeText`. We can then pass the steps and total steps as string there, for example:

```
<BridgeProcessCardTitle badgeText={1/3} title="Unmint your tBTC tokens" />
```

I think this one would be the best because this way, the `BridgeProcessCardTitle` will be more reusable. If we provide `badgeText` it will display the badge with the text on the left side of the title.

Another good apporach would be to create a beforeText prop (similar to `afterDescription` and pass a Badge with steps there. But it looks like this badge is the common thing for those Bridge Process Card Titles and we don't plan to display anything else there, so I would just create `badgeText` for now.

File: src/pages/tBTC/Bridge/components/BridgeProcessCardTitle.tsx [hunk 16]
```diff
@@ -1,50 +1,81 @@
...
+        <BodySm
+          rounded="md"
+          bg="#1E1E1E"
+          p={2}
+          fontWeight="medium"
+          lineHeight={4}
+          color="hsl(182, 100%, 70%)"
+        >
+          {step}/{TOTAL_STEPS}
+        </BodySm>
```
Comment (michalsmiarowski) [inline src/pages/tBTC/Bridge/components/BridgeProcessCardTitle.tsx:66]: I think we can use `Badge` here. It's so similar to `Action off-chain` badge below, that I think it would make more sense:

```suggestion
        <Badge variant="subtle" rounded="md" p={"2"} lineHeight={4}>
          {step}/{TOTAL_STEPS}
        </Badge>
```

**Review by michalsmiarowski — CHANGES_REQUESTED** [review]
Left some more small comments before we merge

File: src/pages/tBTC/Bridge/Minting/ProvideData.tsx
Comment (kpyszkowski) [reply src/pages/tBTC/Bridge/Minting/ProvideData.tsx:1]: Ref commit: feec30c6

File: src/pages/tBTC/Bridge/Minting/ProvideData.tsx (hunk 13 above)
Comment (kpyszkowski) [reply src/pages/tBTC/Bridge/Minting/ProvideData.tsx]: Ref commit: 73c8aadb

File: src/pages/tBTC/Bridge/Minting/ProvideData.tsx (hunk 14 above)
Comment (kpyszkowski) [reply src/pages/tBTC/Bridge/Minting/ProvideData.tsx]: Ref commit: e4955c23

File: src/pages/tBTC/Bridge/Unmint.tsx (hunk 15 above)
Comment (kpyszkowski) [reply src/pages/tBTC/Bridge/Unmint.tsx]: Ref commit: 5e281add

File: src/pages/tBTC/Bridge/components/BridgeProcessCardTitle.tsx (hunk 16 above)
Comment (kpyszkowski) [reply src/pages/tBTC/Bridge/components/BridgeProcessCardTitle.tsx]: Ref commit: 5e281add

File: src/pages/tBTC/Bridge/Unmint.tsx (hunk 15 above)
Comment (michalsmiarowski) [reply src/pages/tBTC/Bridge/Unmint.tsx]: Looking good, although ideally we would want to make `badgeText` optional like I mentioned here:

> If we provide titleBadgeText it will display the badge with the text on the left side of the title, if not, the badge will not be displayed.

We can do it in the future though when we decide we need it (and we probably will for resume deposit, but who knows).

As for now I am resolving this conversation

**Review by michalsmiarowski — APPROVED** [review]
LGTM 🔥
